File: ruby/hamming/hamming.rb
class Hamming

  def self.compute(one, two)
    self.compare(one, two) unless self.different_strings(one, two)
  end

  def self.compare(one, two)
    control, compare, distance = one.split(//), two.split(//), 0
    control.each_with_index { |v, i| distance += 1 if compare[i] != v }

    distance
  end

  def self.different_strings(one, two)
    if one.length != two.length then raise ArgumentError else false end
  end

end

module BookKeeping
  VERSION = 3
end

File: javascript/hamming/hamming.js
var Hamming = function() {};

Hamming.prototype.compute = function(one, two) {
  if (one.length !== two.length) {
    throw 'DNA strands must be of equal length.';
  } else {
    return compare(one, two);
  }
}

function compare(one, two) {
  var string_one = one.split('');
  var string_two = two.split('');
  var distance = 0;

  for(var i = 0; i < string_one.length; i++) {
    if (string_one[i] !== string_two[i]) { distance++; }
  }

  return distance;
}

module.exports = Hamming;
